components/molecules/charts/schedule/helper/columns.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * Components - Molecules - Charts - Schedule - Helper - Columns
+ */
+
+// React
+import React from 'react'
+
+// Style
+import styled from 'styled-components'
+
+// Date FNS
+import {
+  endOfMonth,
+  endOfDay,
+  endOfWeek,
+  startOfMonth,
+  startOfWeek,
+  startOfDay,
+  setMonth,
+  setDay
+} from 'date-fns'
+
+// UI
+import ColumnPagination from './columnPagination'
+import Tooltip from '../../../../atoms/tooltip/tooltip'
+import DATE_TYPE from '../../../../constants/dateType'
+
+const COLUMN_PATTERN = {
+  day: [...Array(12).keys()],
+  month: ['Week1', 'Week2', 'Week3', 'Week4'],
```

Comment:
@Ojaghi1991 There can sometimes be 5 weeks in a month - be careful with that, test this works in such occassions - also what day or the week does your week start

Reply:
@iwarner handled with eventTimeSplitting props and listen to that and used  these methods 
`  endOfMonth,
  endOfDay,
  endOfWeek,
  startOfMonth,
  startOfWeek,
  startOfDay,
`